src/app/src/sensorUtils.js

```diff
@@ -17,46 +29,87 @@ export function makeRiverGaugeRequest(id, isApiRequest) {
     return axios.get(url);
 }
 
-function parseCsvString(csvString) {
-    const data = Papa.parse(csvString, { header: true, comments: "#", dynamicTyping: true, skipEmptyLines: true });
-    return data;
+export function parseCsvString(csvString) {
+    const data = Papa.parse(csvString, {
+        header: true,
+        comments: '#',
+        dynamicTyping: true,
+        skipEmptyLines: true,
+    });
+
+    return data.data;
 }
 
-export function parseRiverGaugeApiData(id, apiData) {
-    const {
-        data: {
-            value: {
-                timeSeries: data,
-            },
+export function parseRiverGaugeApiData(id, data) {
+    const extractedVariableData = VARIABLES.reduce(
+        (acc, variable, idx) => {
+            const apiVariableData = data[idx];
+            if (apiVariableData) {
+                const sensorValue = Number(
+                    apiVariableData.values[0].value[0].value
+                );
+                return sensorValue !== apiVariableData.variable.noDataValue
+                    ? Object.assign(acc, { [variable]: sensorValue })
+                    : Object.assign(acc, { [variable]: 0 });
+            }
+            return acc;
         },
-    } = apiData;
-
-    const extractedVariableData = VARIABLES.reduce((acc, variable, idx) => {
-        const apiVariableData = data[idx];
-        if (apiVariableData) {
-            const sensorValue = Number(apiVariableData.values[0].value[0].value);
-            return sensorValue !== apiVariableData.variable.noDataValue
-                ? Object.assign(acc, {[variable]: sensorValue })
-                : Object.assign(acc, {[variable]: 0 });
-        }
-        return acc;
-    }, { id });
+        { id }
+    );
 
     return extractedVariableData;
 }
 
-export function parseRiverGaugeCsvData(id, csvString) {
-    const {
-        data,
-    } = csvString;
-
-    const parsedData = parseCsvString(data);
-    const dataRow = parsedData.data.slice(-1)[0];
+export function parseRiverGaugeCsvData(id, data) {
+    const dataRow = data.slice(-1)[0];
 
-    const extractedVariableData = VARIABLES.reduce((acc, variable) => {
-        const code = `p${VARIABLE_CODES[variable]}`;
-        return Object.assign(acc, {[variable]: dataRow[code] || 0 });
-    }, { id });
+    const extractedVariableData = VARIABLES.reduce(
+        (acc, variable) => {
+            const code = `p${VARIABLE_CODES[variable]}`;
+            return Object.assign(acc, { [variable]: dataRow[code] || 0 });
+        },
+        { id }
+    );
 
     return extractedVariableData;
 }
+
+export function getSensorByProp(prop, value) {
+    const sensor = find(sensors.features, f => {
+        return f.properties[prop] === value;
+    });
+
+    return sensor;
```

Comment:
If we'd like to shorten this a little I think it could be:

```
export getSensorByProps = (prop, value) => find(sensors.features, ({ properties }) => properties[prop] === value);
```